### ISSUE TITLE
problems with readonly elements

### ISSUE THREAD

**Issue by My1** [twbs/bootstrap#16413]
I found 2 things.
1 you can mark a checkbox div as disabled, but not as readonly. I am not sure whether or not you can set the div as disabled without affecting screenreaders and similar.
2 a readonly input within a form-control gets the not-allowed cursor, even though the docs say that readonly elements are supposed to keep the normal cursor.

**Comment by patrickhlauke** [comment]
Please provide some simplified test cases (e.g. using http://jsfiddle.net/ or similar) to illustrate your concerns. One immediate thought, though:

- if you're making a checkbox out of a `<div>`, there are probably more pressing concerns here for accessibility/screenreaders (such as adding the correct `role="checkbox"`, `tabindex=0` to make it focusable, dynamically setting `aria-checked="true"`/`aria-checked="false"`); so first of all, i'd say: if you're making a checkbox, use an `<input type="checkbox" ...>`

**Comment by My1** [comment]
WRONG!
I have an input type=checkbox WITHIN a checkbox div, similar to the docs, but with readonly.
keeping it short:
```
<div class="checkbox">
  <label>
    <input type="checkbox" value="">
    Option one is this and that&mdash;be sure to include why it's great
  </label>
</div>
<div class="checkbox disabled">
  <label>
    <input type="checkbox" value="" readonly>
    Option two is disabled
  </label>
</div>
```
seemingly I can tab the checkboxes I have.

**Comment by patrickhlauke** [comment]
> WRONG!

firstly, can I suggest you change your tone, as we're only trying to help, and your original description of the problem was far from clear...

So then:

> you can mark a checkbox div as disabled, but not as readonly.

Yes. Expanding on your statement, is your question: "could we also have a `.readonly` class which, in the case of checkboxes/radio buttons, keeps the cursor from turning into a hand/activation cursor when hovering over the `<label>`?"

> I am not sure whether or not you can set the div as disabled without affecting screenreaders and similar

simply setting a class of `.disabled` won't prevent keyboard/screenreader users from being able to focus and activate native form controls (in most, if not all, cases). So, as per the examples in http://getbootstrap.com/css/#checkboxes-and-radios and http://getbootstrap.com/css/#forms-control-disabled individual form controls that are to be disabled need to actually have the `disabled` *attribute* as well.

> a readonly input within a form-control gets the not-allowed cursor, even though the docs say that readonly elements are supposed to keep the normal cursor

assuming this still relates to the example code you gave above, the cursor is coming from the fact that you have the `.disabled` class on the `<div>` that wraps around the `readonly` checkbox.

**Comment by My1** [comment]
okay the thing with the checkbox as a div was so wrong I didnt even know that was possible (I am not really one with javascript), and your docs also put a div.checkbox around it, so I was pretty confused. I didnt mean to shout but I have no Idea how to bold here, I an usually a bbcode user.

and the readonly is not together with a css this rather relates to this little thing which is really a bit of a confusio. coming from my side.
look at this:
```HTML
<div class="form-group readonly">
      <label for="title">Titel</label>
      <input class="form-control" type="text" name="title" id="title" readonly value="nothing special" style="margin-left: auto;margin-right: auto;min-width:18em;width:50%;">
      </div>
```
this thing immediately gets a not allowed cursor because of said lines in CSS.
I have a little something against that.
```CSS
.form-control[readonly] {
cursor:auto;
}
```

**Comment by patrickhlauke** [comment]
> this thing immediately gets a not allowed cursor

I'm not seeing that http://jsfiddle.net/brywree1/1/ - sure there's no other classes further up in your document interferring?

**Comment by twbs-lmvtfy** [comment]
Hi @patrickhlauke!

You appear to have posted a live example (http://fiddle.jshell.net/brywree1/1/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 34, column 3 thru line 35, column 1: Saw ```` < ```` when expecting an attribute name. Probable cause: Missing ```` > ```` immediately before.
* line 34, column 3 thru line 35, column 33: Attribute ```` <div ```` not allowed on element ```` form ```` at this point.
* line 38, column 7 thru line 38, column 12: Stray end tag ```` div ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by My1** [comment]
okay I checked something.I have an old CSS, I have it locally, seems it was 3.3.2 and it was done already. (lol)
and okay with the checkboxes essentially I just have to do a div.checkbox.disabled > checkbox[readonly] right?

**Comment by patrickhlauke** [comment]
> but what I dont like is the disabled cursor...

well, if you *do* want to add the `.disabled` class to the outer `<div>` but then *don't* want the disabled cursor, then yes, you'll need to override it yourself with something like

```
div.checkbox.disabled > input[readonly] { cursor: auto; }
```
